src/huggingface_hub/hf_api.py

```diff
@@ -786,6 +795,8 @@ def list_datasets(
         if full is not None:
             if full:
                 params.update({"full": True})
+        if cardData is not None:
+            params.update({"cardData": True})
```

Comment:
I'd apply the same as for the `full`:
```suggestion
        if cardData is not None:
            if cardData:
                params.update({"cardData": True})
```

This way, if it's passed as `False` then it's still handled correctly.